File: TypeScript/Math/get-smallest-int-in-slice.ts
function main(): void {
    // Create a slice of ints
    let randomValues: number[] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
    // Get the smallest int in the slice
    console.log(getSmallestIntInSlice(randomValues))
}

main()

// Get the smallest int in a slice of ints
function getSmallestIntInSlice(slice: number[]): number {
    let smallestInt: number = slice[0]
    for (let i: number = 1; i < slice.length; i++) {
        if (slice[i] < smallestInt) {
            smallestInt = slice[i]
        }
    }
    return smallestInt
}

File: Python/Math/get_smallest_int_in_slice.py
# Get the smallest int in a given slice.
def get_smallest_int_in_slice(given_slice):
    return min(given_slice)


def main():
    # Create a slice of ints from 1 to 10.
    randomSlice = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
    # Get the smallest int in the slice.
    print(get_smallest_int_in_slice(randomSlice))


main()
